src/cudnn/tensor/descriptor.rs

```diff
@@ -0,0 +1,66 @@
+use super::super::sys::*;
+use crate::prelude::*;
+use core::marker::PhantomData;
+use core::mem::MaybeUninit;
+
+pub(crate) const TENSOR_FORMAT: cudnnTensorFormat_t = cudnnTensorFormat_t::CUDNN_TENSOR_NCHW;
+/// A descriptor of a tensor. This can be reused as all tensors are fully packed
+/// and in NCHW layout.
+///
+/// This descriptor is destroyed when it is dropped.
+///
+/// # See also
+/// <https://docs.nvidia.com/deeplearning/cudnn/api/index.html#cudnnTensorDescriptor_t>
+/// <https://docs.nvidia.com/deeplearning/cudnn/api/index.html#cudnnDestroyTensorDescriptor>
+pub struct TensorDescriptor<T, const N: usize, const C: usize, const H: usize, const W: usize> {
```

Comment:
We also need to support up to 6d tensors, which it looks like you can do pretty easily with the tensor descriptors, this can be tracked at runtime like everything else 😄

Reply:
so TensorNd = TensorDescriptorNd([Axes]) + CudaSlice?